core/include/traccc/seeding/doublet_finding.hpp

```diff
@@ -11,35 +11,40 @@
 #include "traccc/seeding/detail/doublet.hpp"
 #include "traccc/seeding/detail/singlet.hpp"
 #include "traccc/seeding/detail/spacepoint_grid.hpp"
+#include "traccc/seeding/detail/spacepoint_type.hpp"
 #include "traccc/seeding/doublet_finding_helper.hpp"
 #include "traccc/utils/algorithm.hpp"
 
 namespace traccc {
 
 /// Doublet finding to search the combinations of two compatible spacepoints
+/// @tparam otherSpType is whether it is for middle-bottom or middle-top doublet
+template <details::spacepoint_type otherSpType>
 struct doublet_finding
     : public algorithm<std::pair<doublet_collection_types::host,
                                  lin_circle_collection_types::host>(
-          const sp_grid&, const sp_location&, const bool&)> {
+          const sp_grid&, const sp_location&)> {
 
     /// Constructor for the doublet finding
     ///
     /// @param seedfinder_config is the configuration parameters
     /// @param isp_container is the internal spacepoint container
-    doublet_finding(const seedfinder_config& config) : m_config(config) {}
+    doublet_finding(const seedfinder_config& config) : m_config(config) {
+        static_assert(otherSpType == details::spacepoint_type::bottom ||
+                      otherSpType == details::spacepoint_type::top);
```

Comment:
At this point I'm just using this PR as a teaching tool, this change is not really all that important: In this case the struct itself should never be instantiated with `traccc::details::spacepoint_type::middle`. So the `static_assert(...)` should also be put at the "struct level". Like we do here for instance:

https://github.com/acts-project/vecmem/blob/main/core/include/vecmem/containers/data/jagged_vector_buffer.hpp#L44-L52

Reply:
What I mean is that it should **only** be put at the "struct level". (Re-reading now I see that I wrote it in a very misunderstand-able way.)

Reply:
I see, I wasn't aware this was possible.